src/briefcase/integrations/docker.py

```diff
@@ -189,16 +212,156 @@ def _buildx_installed(cls, tools: ToolCache):
+        inside the container to be owned on the host file system by the user running
+        Docker. Additionally, though, because the host user is mapped to root inside the
+        container, any files that were created by the host user in the bind mount
+        outside the container are owned by root inside the container; therefore, a step-
+        down user could not interact with such bind mount files inside the container.
+
+        To accommodate these different modes, this checks which user owns a file that is
+        created inside a bind mount in the container. If the owning user of that file on
+        the host file system is root, then a step-down user is necessary inside
+        containers. If the owning user is the host user, root should be used.
+
+        On macOS, Docker Desktop is the only option to use Docker and user mapping
+        happens differently such that any user in the container is mapped to the host
+        user. Instead of leveraging user namespaces as on Linux, this user mapping
+        manifests as a consequence of bind mounts being implemented as NFS shares
+        between macOS and the Linux VM that Docker Desktop runs containers in. So,
+        using a step-down user on macOS is effectively inconsequential.
+
+        On Windows WSL 2, Docker Desktop operates similarly to how it does on Linux.
+        However, user namespace mapping is not possible because the Docker Desktop VM
+        and the WSL distro are already running in different user namespaces...and
+        therefore, Docker cannot even see the users in the distro to map them in to the
+        container. So, a step-down user is always used.
+
+        ref: https://docs.docker.com/engine/security/userns-remap/
+
+        :param image_tag: The image:tag to use to create the container for the test; if
+            one is not specified, then `alpine:latest` will be used.
+        :returns: True if users are being mapped; False otherwise
+        """
+        host_write_test_path = self._write_test_path()
+        container_write_test_path = PurePosixPath(
+            "/host_write_test", host_write_test_path.name
+        )
+
+        docker_run_cmd = [
+            "docker",
+            "run",
+            "--rm",
+            "--volume",
+            f"{host_write_test_path.parent}:{container_write_test_path.parent}:z",
+            "alpine" if image_tag is None else image_tag,
+        ]
+
+        host_write_test_path.parent.mkdir(exist_ok=True)
+
+        try:
+            host_write_test_path.unlink(missing_ok=True)
+        except OSError as e:
+            raise BriefcaseCommandError(
+                f"""\
+The file path used to determine how Docker is mapping users between the host
+and Docker containers already exists and cannot be automatically deleted.
+
+    {host_write_test_path}
+
+Delete this file and run Briefcase again.
+"""
+            ) from e
+
+        try:
+            self.tools.subprocess.run(
```

Comment:
An optimization that I've considered by didn't implement is leveraging `docker exec`.

Here, as in other places, we're spinning up an entire container for one quick command...just to immediately destroy the entire thing. Alternatively, we could start the container at the beginning and just call `docker exec` to run commands ad hoc and then destroy the container when the command is finishing.

It may even be possible to allow multiple commands to use the same container.

I haven't done much assessment on what this would take or consequences from reusing the same container....but in my virtualized environments and my Pis, all this docker setup and teardown definitely slows things down.